stock_scheduler_assignation_horizon/migrations/14.0.1.0.1/pre-migration.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2022 Camtocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import SUPERUSER_ID, api
+from odoo.tools.sql import column_exists
+
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    if column_exists(cr, "res_company", "is_moves_assignation_limited"):
```

Comment:
You can count on the column to exist, because it did before this new module version.
If the migration is executed, it means you're coming from a previous version: the column definitely exists

In any case, it doesn't hurt to check I guess.

I would avoid the nested `if` indentation by reverting your conditions.

```suggestion
    if not column_exists(cr, "res_company", "is_moves_assignation_limited"):
        return
    # ...
```

Alternatively, since the field already exist.. I think you can simply use the env rather than sql queries

```py
    if "is_moves_assignation_limited" not in env["res.company"]._fields:
        return
    companies = env["res.company"].search([])

    stock_horizon_move_assignation = any(c.is_moves_assignation_limited for c in companies)
    if not stock_horizon_move_assignation:
        return
    _create_ir_config_param(
        "stock_scheduler_assignation_horizon.stock_horizon_move_assignation",
        True,
        "stock_scheduler_assignation_horizon.is_moves_assignation_limited",
    )

    stock_horizon_move_assignation_limit = min(
        c.moves_assignation_horizon for c in companies
        if c.moves_assignation_horizon,
        default=0
    )
    if not stock_horizon_move_assignation_limit:
        return
    _create_ir_config_param(
       "stock_scheduler_assignation_horizon.stock_horizon_move_assignation_limit",
        stock_horizon_move_assignation_limit,
        "stock_scheduler_assignation_horizon.stock_scheduler_assignation_horizon",
    )

def _create_ir_config_param(key, value, xmlid):
    # ...
```